### PR TITLE
fix the pipeline

### PR DESCRIPTION
**Pull request by adrinjalali**
I fixed the pipeline and included an imputer in it, which makes it quite nice IMO. And I think you could put this whole pipeline in a fairlearn.GridSearch if you want, so it'd be quite nice in other examples too.

I've also added a setting to tell black to respect 79 character limit for lines, and the code is formatted with black.

Merging this to your branch will show up in your PR :)

WDYT @riedgar-ms?

### CODE REVIEW

File: examples/plot_make_derived_metric.py [hunk 1]
```diff
@@ -82,47 +79,36 @@
...
+    steps=[
+        ("impute", SimpleImputer()),
+        ("scaler", StandardScaler()),
+    ]
+)
+categorical_transformer = Pipeline(
     [
-        ("keep_cols", "passthrough", pass_through_columns),
-        ("ohe_cols", OneHotEncoder(sparse=False, handle_unknown='ignore'), ohe_columns)
+        ("impute", SimpleImputer(strategy="most_frequent")),
```
Comment (riedgar-ms) [inline examples/plot_make_derived_metric.py:90]: I suppose this is a case where there's no good solution. It's just.... filling in _anything_ which is unknown feels statistically dubious to me (the ML model will then be biased by however the imputation stage worked). But dropping rows which have missing data will produce other biases :-/

File: examples/plot_make_derived_metric.py [hunk 2]
```diff
@@ -82,47 +79,36 @@
...
+        ("ohe", OneHotEncoder(handle_unknown="ignore")),
+    ]
+)
+preprocessor = ColumnTransformer(
+    transformers=[
+        ("num", numeric_transformer, selector(dtype_exclude="category")),
+        ("cat", categorical_transformer, selector(dtype_include="category")),
+    ]
+)
+unmitigated_predictor = Pipeline(
```
Comment (riedgar-ms) [inline examples/plot_make_derived_metric.py:100]: Thanks for the sample of how to do this!

File: examples/plot_make_derived_metric.py [hunk 3]
```diff
@@ -82,47 +79,36 @@
...
+    steps=[
+        ("preprocessor", preprocessor),
+        (
+            "classifier",
+            LogisticRegression(solver="liblinear", fit_intercept=True),
```
Comment (riedgar-ms) [inline examples/plot_make_derived_metric.py:105]: Not for this, but for our mitigation algorithms.... can we pass in `sample_weight` and get it to the right place? :-)

File: examples/plot_make_derived_metric.py (hunk 1 above)
Comment (riedgar-ms) [reply examples/plot_make_derived_metric.py:90]: Would you object if I added a comment along the lines of "be careful" around using the imputer? I don't want to get bogged down when it's not the main subject of this demonstration, but I feel that it is another potentially problematic operation.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[tool.black]
```
Comment (riedgar-ms) [inline pyproject.toml:1]: I know you're advocating this, and I have no objection in general, but.... including it is orthogonal to this changeset.

Comment (riedgar-ms) [reply pyproject.toml:1]: For this purpose, I can just remove it when it gets to my branch. For now, my big concern is about the imputation.

File: examples/plot_make_derived_metric.py (hunk 1 above)
Comment (adrinjalali) [reply examples/plot_make_derived_metric.py:90]: Sure, people do use these simple techniques though. I wouldn't be too worried about it. But we are working on better imputation techniques. Feel free to leave the warning comment around it :)

File: examples/plot_make_derived_metric.py (hunk 3 above)
Comment (adrinjalali) [reply examples/plot_make_derived_metric.py:105]: yes, you can pass the `sample_weight` to a step of the pipeline with the name of the step, so it'd be `classifier__sample_weight` in this case, and it'll be passed to the `fit` method of the classifier. In the near future we'd be able to handle this much better with the sample props proposal implementation.

File: pyproject.toml (hunk 4 above)
Comment (adrinjalali) [reply pyproject.toml:1]: Sure, but please note that this doesn't do anything, it just tells `black` to use this option if the user runs black on something. Feel free to leave it out, I'll add it in a separate PR.

File: examples/plot_make_derived_metric.py (hunk 2 above)
Comment (adrinjalali) [reply examples/plot_make_derived_metric.py:100]: I guess you can tell I do enjoy playing with these pieces like lego :D

**Pull request merged**